### PR TITLE
Adds examine messages to fire/lava/freeze/acid proofed clothing items.

### PR DESCRIPTION
**Pull request by ShizCalev**
:cl: ShizCalev
tweak: Clothing (bags, gloves, belts, uniforms, armor, ect) will now inform you if they are resistant to frost, fire, acid, and lava when examined.
/:cl:

### CODE REVIEW

File: code/__DEFINES/flags.dm [hunk 1]
```diff
@@ -64,6 +64,19 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
...
+/obj/item/proc/clothing_resistance_flag_examine_message(mob/user)
+	if(resistance_flags & INDESTRUCTIBLE)
+		to_chat(user, "[src] seems extremely robust! It'll probably withstand anything that could happen to it!")
+		return
+	if(resistance_flags & FREEZE_PROOF)
+		to_chat(user, "[src] is made of cold-resistant materials.")
+	if(resistance_flags & FIRE_PROOF)
+		to_chat(user, "[src] is made of fire-retardant materials.")
+	if(resistance_flags & LAVA_PROOF)
+		to_chat(user, "[src] is made of an extremely heat-resistance material, it'd probably be able to withstand lava!")
```
Comment (swindly) [inline code/__DEFINES/flags.dm]: heat-resistant

**Comment by Bawhoppen** [comment]
I don't think we should over-inform people with information about items. If people don't know that something is fireproof ingame, why would them looking closely at that item teach them it was? I know it's arbitrary what examining tells you already anyways, but this seems unneeded.

**Comment by Barhandar** [comment]
Because if fire's going to LOLOBLITERATE everything it touches, players _need_ to know whether something is safe or not. Moreso with lava stripping fireproofing magically so potion's namechange isn't indicative.

**Comment by fludd12** [comment]
Lavaproofing shouldnt strip fireproofing, it should just... Not apply fireproofing. If it's actually removing fireproofing, there is a problem.

**Comment by fludd12** [comment]
Oh! *Lava* strips fireproofing, not *lavaproofing.* I can read I swear.

I was like `i coded that potion how is that intended` and then I read the code snippet

**Comment by Bombpants** [comment]
Are lava proof things also considered fire proof?

**Comment by Barhandar** [comment]
Lavaproofing and fireproofing are separate things, while most lavaproof things also have fireproof flag, it's not mandatory. 
@ShizCalev _intended_ does not mean _good_ or _warranted_. Especially since the intention seems to be making them burn up in lava, not outside lava when escaped out of it.

File: code/__DEFINES/flags.dm [hunk 2]
```diff
@@ -64,6 +64,35 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
...
+		to_chat(user, "[src] is made of an extremely heat-resistant material, it'd probably be able to withstand lava!")
+	if(resistance_flags & (ACID_PROOF | UNACIDABLE))
+		to_chat(user, "[src] looks pretty robust! It'd probably be able to withstand acid!")
+	if(resistance_flags & FREEZE_PROOF)
+		to_chat(user, "[src] is made of cold-resistant materials.")
+
+	var/fire_resist_message = "[src] is made of fire-retardant materials."
+	if(resistance_flags & FIRE_PROOF)
+		to_chat(user, fire_resist_message)
+	else if(istype(src, /obj/item/clothing))
```
Comment (SpaceManiac) [inline code/__DEFINES/flags.dm]: works, but `istype(src)` is suspicious design

**Pull request merged**